libkbfs/block_types.go

```diff
@@ -188,10 +188,30 @@ func (fb *FileBlock) NewEmpty() Block {
 
 // DataVersion returns data version for this block.
 func (fb *FileBlock) DataVersion() DataVer {
+	// If this is an indirect block, and none of its children are
+	// marked as direct blocks, then this must be a big file.  Note
+	// that we do it this way, rather than returning on the first
+	// non-direct block, to support appending to existing files and
```

Comment:
I'm a little bit wary of changing the `BlockPointer` for blocks that aren't going to be getting a new `BlockID`, because it will break anywhere that uses a `BlockPointer` as a map key, which is a ton of places.  Probably in most of those places we should be using `BlockID` instead, but I don't really want to do that as part of this PR.